language/vm/vm-runtime/src/interpreter.rs

```diff
@@ -549,6 +549,14 @@ where
                         gas!(const_instr: context, self, Opcodes::XOR)?;
                         self.binop_int(|l: u64, r| Some(l ^ r))?
                     }
+                    Bytecode::Shl => {
+                        gas!(const_instr: context, self, Opcodes::SHL)?;
+                        self.binop_int(|l: u64, r| Some(l << r))?
```

Comment:
What happens if `r` is too big here? Will we get an `ArithmeticError`? Similar question for `Shr`.

Reply:
Good catch and my bad. I had some misunderstanding about the machine shl/shr instructions and how languages like C/C++ handle them. (I thought the de facto standard is to make the result zero instead of panicking, which is wrong.)

I make sure the number of bits shifted is checked here.

Reply:
`make the result zero instead of panicking, which is wrong.` Is it? 

I would have expected it to 0 out because its an arithmetic shift

Reply:
Apparently Rust (in debug) panics on any shift too large, left or right. Perhaps this is the safest behavior?

Reply:
> I would have expected it to 0 out because its an arithmetic shift
@tnowacki I had the same thing in my mind, but after doing some research I think many languages actually don't distinguish arithmetic and logical shift clearly. We need to define this explicitly for move. Is there a reason we should favor ARITHMETIC_ERROR or 0?

As for the << operator with overflowing bits:
Rust panics. 
C/C++ says it's UB.

Tried a few compilers online and a version of clang translates it to shlq while gcc translates it to salq, producing different results.

Reply:
I think specifying (in both name and documentation) whether the right shift is arithmetic or logical is really important, particularly if we add signed integers in the future. Panicking seems more consistent with out handling of other behaviors like overflow.